### PR TITLE
build: replace travis with github actions for automation

### PR DESCRIPTION
**Pull request by davidjoy**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        run: npm run lint
+      - name: Test
+        run: npm run test
+      - name: Release
+        uses: cycjimmy/semantic-release-action@v2
+        with:
+          semantic_version: 16
+        env:
+          GITHUB_TOKEN: ${{ secrets.SEMANTIC_RELEASE_GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.SEMANTIC_RELEASE_NPM_TOKEN }}
```
Comment (stvstnfrd) [inline .github/workflows/release.yml:34]: Did the secrets already exist there or did you have to migrate them?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        fetch-depth: 0
+    - name: Setup Nodejs
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12
+    - name: Install dependencies
+      run: npm ci
+    - name: Lint
+      run: npm run lint
+    - name: Test
```
Comment (stvstnfrd) [inline .github/workflows/ci.yml:19]: It's too bad we can't split these out into separate "jobs" without copying all that boilerplate...

**Review by stvstnfrd — APPROVED** [review]
As long as it works, LGTM

File: .github/workflows/release.yml (hunk 1 above)
Comment (davidjoy) [reply .github/workflows/release.yml:34]: I didn't do anything special there and am not familiar with how the secrets work... should I have had to do something?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (davidjoy) [reply .github/workflows/ci.yml:19]: I just copied these files from frontend-build, to be clear.  Maybe I'm not following the right process?

File: .github/workflows/release.yml (hunk 1 above)
Comment (stvstnfrd) [reply .github/workflows/release.yml:34]: I checked; there do appear to be pre-existing Github secrets by those names, so it _should_ work :crossed_fingers:

File: .github/workflows/ci.yml (hunk 2 above)
Comment (stvstnfrd) [reply .github/workflows/ci.yml:19]: Nah, sorry, you're good.

This works; it runs all the tests in serial and reports status as a single job.
To have multiple job "checks" (eslint, tests, commitlint) each with their own status and run in parallel, one could write this workflow to contain multiple entries in the `jobs` dict (there's currently just one, `build`), but it requires copy/pasting all of that setup boilerplate.. or writing a common, parameterized build script (Makefile, bash, etc.) that handles it.

Again, this is good as written, just lamenting that Github Actions don't appear to have a "common_setup" stage/step.

**Pull request merged**